src/instrumenter/instrument.ts

```diff
@@ -127,6 +134,26 @@ const CHECK_STATE_INVS_FUN = "__scribble_check_state_invariants";
 const OUT_OF_CONTRACT = "__scribble_out_of_contract";
 const CHECK_INVS_AT_END = "__scribble_check_invs_at_end";
 
+function getAllNames(contract: ContractDefinition): Set<string> {
+    const nameSet: Set<string> = new Set();
+    for (const v of contract.getChildren()) {
+        if ("name" in v) nameSet.add(v["name"]);
+    }
+    return nameSet;
+}
+
+function getNamesInFuncScope(contract: ContractDefinition, fn: FunctionDefinition): Set<string> {
```

Comment:
There are several additional categories of declared names, that are missing here:

 - structs, events and enums declared in the current contract
 - structs, enums and constants declared at the source unit level
 - names that are imported (and potentially renamed) from other files.
 
Getting the list of all declared names in a function scope, (or any scope) is thus a little tricky :(. We have some code that tries to compute this in `maru-ir`. The more I think about it, perhaps its better we relocate that code to `solc-typed-ast`, and rebase `scribble` to use it.